packages/polkadart_scale_codec/test/test_types/struct_test.dart

```diff
@@ -0,0 +1,181 @@
+import 'package:polkadart_scale_codec/polkadart_scale_codec.dart';
+import 'package:test/test.dart';
+
+void main() {
+  // Creates the registry for parsing the types
+  final registry = TypeRegistry(
+    types: {
+      'OrderJuiceEnum': {
+        '_enum': ['Orange', 'Apple', 'Kiwi']
+      },
+      'OuncesEnum': {
+        '_struct': {'ounces': 'u8', 'Remarks': 'Option<Text>'}
+      },
+      'OrderStruct': {
+        '_struct': {
```

Comment:
Shouldn't we test all supported types with struct? 
```suggestion
        '_struct': {
        'index': 'u8',
          'note': 'Text',
          'Juice': 'OrderJuiceEnum',
          'Ounces': 'OuncesEnum',
          'someVec': 'Vec<u8>',
          'someBool': 'bool',
```

Reply:
Or maybe only testing the primitive is enough. I'm not sure